invenio/modules/access/mailcookie.py

```diff
@@ -126,7 +126,7 @@ def mail_cookie_check_role(cookie, uid):
     except (TypeError, AssertionError, StandardError):
         raise InvenioWebAccessMailCookieError
     expiration = (datetime.today()+role_timeout).strftime(_datetime_format)
-    acc_add_user_role(uid, role_id, expiration)
+    acc_add_user_role(uid, role_id, expiration=expiration)
```

Comment:
Without the named argument the expiration date was matched to the e-mail address, because of the paramter order in the function definition